imageio/plugins/rawpy.py

```diff
@@ -0,0 +1,187 @@
+
+        self._request.finish()
+
+    def read(self, *, index: int = 0, **kwargs) -> np.ndarray:
+        """Read Raw Image.
+
+        Returns
+        -------
+        nd_image: ndarray
+            The image data
+        """
+
+        nd_image: np.ndarray
+
+        try:
+            nd_image = self._image_file.postprocess(**kwargs)
+        except Exception as ex:
+            raise ex
+
+        return nd_image
+
+    def write(
+        self, 
+        ndimage: Union[ArrayLike, List[ArrayLike]]
+    ) -> bytes | None:
+        """RawPy does not support writing.
+        """
+        raise NotImplementedError()
+
+    def iter(self) -> Iterator[np.ndarray]:
+        """Load the image.
+
+        Returns
+        -------
+        nd_image: ndarray
+            The image data
+        """
+
+        try:
+            yield self.read()
+        except Exception as ex:
+            raise ex
+
+    def metadata(
+        self, index: int = None, exclude_applied: bool = True
+    ) -> Dict[str, Any]:
+        """Read ndimage metadata.
+
+        Parameters
+        ----------
+        exclude_applied : bool
+            If True, exclude metadata fields that are applied to the image while
+            reading. For example, if the binary data contains a rotation flag,
+            the image is rotated by default and the rotation flag is excluded
+            from the metadata to avoid confusion.
+
+        Returns
+        -------
+        metadata : dict
+            A dictionary of format-specific metadata.
+
+        """
+
+        metadata = {}
+
+        ImageSize = self._image_file.sizes
+
+        # TO DO: Find out what information needs to be included below and refine accordingly.
+        metadata["black_level_per_channel"] = self._image_file.black_level_per_channel
+        metadata["camera_white_level_per_channel"] = self._image_file.camera_white_level_per_channel
+        metadata["color_desc"] = self._image_file.color_desc
+        #metadata["color_matrix"] = self._image_file.color_matrix
+        metadata["daylight_whitebalance"] = self._image_file.daylight_whitebalance
+        metadata["dtype"] = self._image_file.raw_image.dtype.type
+        metadata["flip"] = ImageSize.flip
+        metadata["num_colors"] = self._image_file.num_colors
+        #metadata["raw_colors"] = self._image_file.raw_colors
+        #metadata["raw_colors_visible"] = self._image_file.raw_colors_visible
+        #metadata["raw_image"] = self._image_file.raw_image
+        #metadata["raw_image_visible"] = self._image_file.raw_image_visible
+        #metadata["raw_pattern"] = self._image_file.raw_pattern
+        metadata["raw_type"] = self._image_file.raw_type.name
+        #metadata["rgb_xyz_matrix"] = self._image_file.rgb_xyz_matrix
+        #metadata["tone_curve"] = self._image_file.tone_curve
+        metadata["width"] = ImageSize.width
+        metadata["height"] = ImageSize.height
+        metadata["shape"] = (ImageSize.width, ImageSize.height)
```

Comment:
```suggestion
        metadata["width"] = ImageSize.width
        metadata["height"] = ImageSize.height
```

You don't need to build/combine metadata into more complex types here. It might be nice to in the future if we get a lot of feedback that this is expected and/or something users do themselves a lot. However, I would start with the simplest setup, which is to pass the fields as-is.

Reply:
Thank you for the comment. I do not have much idea about the best practices here. So I started including them and thought I would later remove it if I receive any comment. I have updated the file accordingly now.